plugins/events/create.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const boom = require('boom');
+const urlLib = require('url');
+const validationSchema = require('screwdriver-data-schema');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/events',
+    config: {
+        description: 'Create and start a event',
+        notes: 'Create and start a specific event',
+        tags: ['api', 'events'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const eventFactory = request.server.app.eventFactory;
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const userFactory = request.server.app.userFactory;
+            const scm = eventFactory.scm;
+            const pipelineId = request.payload.pipelineId;
+            const scmContext = request.auth.credentials.scmContext;
+            const startFrom = request.payload.startFrom;
+            const username = request.auth.credentials.username;
+            const payload = {
+                pipelineId,
+                scmContext,
+                startFrom,
+                type: 'pipeline',
+                username
+            };
+            // Match PR-prNum, then extract prNum
+            const prNumFullName = startFrom.match(validationSchema.config.regex.PR_JOB_NAME);
+            const prNum = prNumFullName ? prNumFullName[1].split('-')[1] : null;
+
+            // Fetch the job and user models
+            return Promise.all([
+                pipelineFactory.get(pipelineId),
+                userFactory.get({ username, scmContext })
+            ])
+                // Get scmUri
+                .then(([pipeline, user]) =>
+                    user.getPermissions(pipeline.scmUri)
+                        // Check if user has push access
+                        .then((permissions) => {
+                            if (!permissions.push) {
+                                throw boom.unauthorized(`User ${username} `
+                            + 'does not have push permission for this repo');
+                            }
+                        })
+                        // User has good permissions, create an event
+                        .then(() => user.unsealToken())
+                        .then((token) => {
+                            const scmConfig = {
+                                prNum,
+                                scmContext,
+                                scmUri: pipeline.scmUri,
+                                token
+                            };
+
+                            // Get commit sha
+                            return scm.getCommitSha(scmConfig)
+                                .then((sha) => {
+                                    // For PRs
+                                    if (prNum) {
```

Comment:
what will trigger this flow ? Is'nt all PR event triggering happening via webhooks ?